tests/beta/test_default_tensor_tf.py

```diff
@@ -499,34 +494,46 @@ def circuit(x):
         expected = np.cos(y) ** 2 - np.sin(x) * np.sin(y) ** 2
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
-        res = circuit.jacobian([p])
+        res = qml.jacobian(circuit)(p)
         expected = np.array(
             [-np.cos(x) * np.sin(y) ** 2, -2 * (np.sin(x) + 1) * np.sin(y) * np.cos(y), 0,]
         )
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
-    @pytest.mark.parametrize("diff_method", ["parameter-shift", "finite-diff", "device"])
-    def test_jacobian_agrees(self, diff_method, torch_support, rep, tol):
-        """Test that qnode.jacobian applied to the tensornet.tf device
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "finite-diff", "backprop"])
+    def test_gradient_agrees(self, diff_method, torch_support, rep, tol):
+        """Test that gradient applied to the tensornet.tf device
         returns the same result as default.qubit."""
-        p = np.array([0.43316321, 0.2162158, 0.75110998, 0.94714242])
+        import tensorflow as tf
+
+        p = tf.Variable([0.43316321, 0.2162158, 0.75110998, 0.94714242])
 
         def circuit(x):
-            for i in range(0, len(p), 2):
+            for i in range(0, len(1.0 * p), 2):
```

Comment:
? :thinking:

Reply:
This is because `tf.Variable` isn't iterable, but `tf.Tensor` is --- multiplying by 1.0 is a shortcut to convert a Variable to a Tensor 😆

Reply:
```suggestion
            # multiple the Variable by 1.0 to make it iterable
            for i in range(0, len(1.0 * p), 2):
```